zeus/device/soc/jetson.py

```diff
@@ -0,0 +1,272 @@
+            args=(self.command_queue, self.result_queue, self.power_measurement),
+        )
+        self.process.start()
+
+        atexit.register(self._stop_process)
+
+    def _discover_metrics_and_paths(self):
+        metrics = {}
+        # todo: cite documentation for ina3221
+        path = Path("/sys/bus/i2c/drivers/ina3221x")
+
+        def extract_directories(path, rail_name, rail_index, type):
+            rail_name_lower = rail_name.lower()
+
+            if "cpu" in rail_name_lower:
+                rail_name_simplified = "cpu"
+            elif "gpu" in rail_name_lower:
+                rail_name_simplified = "gpu"
+            elif (
+                "system" in rail_name_lower
+                or "_in" in rail_name_lower
+                or "total" in rail_name_lower
+            ):
+                rail_name_simplified = "total"
+            else:
+                return  # skip unsupported rail types
+
+            if type == "label":
+                power_path = path / f"power{rail_index}_input"
+                volt_path = path / f"in{rail_index}_input"
+                curr_path = path / f"curr{rail_index}_input"
+            else:
+                power_path = path / f"in_power{rail_index}_input"
+                volt_path = path / f"in_voltage{rail_index}_input"
+                curr_path = path / f"in_current{rail_index}_input"
+
+            if check_file(power_path):
+                metrics[rail_name_simplified] = {"power": Path(power_path)}
+            elif check_file(volt_path) and check_file(curr_path):
+                sub = {}
+                sub["volt"] = Path(volt_path)
+                sub["curr"] = Path(curr_path)
+                metrics[rail_name_simplified] = sub
+            else:
+                raise ValueError(
+                    "Not enough measurement data to obtain power readings."
+                )
+
+        for device in path.glob("*"):
+            for subdevice in device.glob("*"):
+                # Get the files containing rail names.
+                label_files = subdevice.glob("in*_label")
+                rail_files = subdevice.glob("rail_name_*")
+                # For each rail name, get its respective power, voltage, current paths.
+                for label_file in label_files:
+                    rail_name = label_file.read_text().strip()
+                    rail_index = label_file.name.split("_")[0].lstrip("in")
+                    extract_directories(subdevice, rail_name, rail_index, "label")
+                for rail_file in rail_files:
+                    rail_name = rail_file.read_text().strip()
+                    rail_index = rail_file.name.split("rail_name_", 1)[-1]
+                    extract_directories(subdevice, rail_name, rail_index, "rail_name")
+        return metrics
+
+    def getAvailableMetrics(self) -> Set[str]:
+        """Return a set of all observable metrics on the Jetson device."""
+        return set(self.metric_paths.keys())
```

Comment:
I'd like this to return metric names with units in them. For instance, `"cpu_mj"`, `"gpu_mj"`, etc.